README.md

```diff
@@ -1 +1,661 @@
+  - [API][api]
+  - [CLIENT][client]
+- the slides will go through each step and present how some the code changes can be created with command line tools
+- some slides will contain a context quote to help keep tract the expected state of the terminal:
+
+> Context: `/graphql-training && docker-compose up -d`
+
+[api]: https://github.com/wunderdogsw/graphql-training/pull/1/commits
+[client]: https://github.com/wunderdogsw/graphql-training/pull/2/commits
+
+---
+
+# Preassignment
+
+---
+
+## Install Nest
+
+Install [NestJS CLI][nestjscli] globally
+
+```plaintext
+yarn global add @nestjs/cli
+```
+
+[nestjscli]: [https://docs.nestjs.com/cli/overview]
+
+Clone the repo and checkout code in the `starter` branch:
+
+```plaintext
+git clone https://github.com/wunderdogsw/graphql-training.git
+cd graphql-training
+git checkout starter
+git checkout -b local
+```
+
+---
+
+Create an API using [NestJS][nestjs]:
+
+```plaintext
+nest new api --package-manager yarn
+```
+
+Create a web app using [CRA][cra]:
+
+```plaintext
+yarn create react-app web --template typescript
+```
+
+
+---
+
+Use the configuration provided in `docker-compose.yml` to start the API, the web app and the DB:
+
+```plaintext
+docker-compose up -d
+```
+
+[nestjs]: https://docs.nestjs.com/
+[cra]: https://create-react-app.dev/docs/getting-started/
+
+Check that the API works at http://localhost:8000
+
+Check that the web app works at http://localhost:3000
+
+Feel free to also check that the DB works at http://localhost:5432 
+
+
+Troubleshooting and shutdown:
+```plaintext
+docker ps
+docker-compose logs api
+docker-compose logs web
+docker-compose logs db
+docker-compose down
+```
+---
+
+# API
+
+> Context: `/graphql-training && docker-compose up -d`
+
+Check that the API works at http://localhost:8000
+
+---
+
+## Create simple GraphQL API
+
+> Context: `/graphql-training/api`
+
+Add GraphQL to the API
+
+```plaintext
+yarn add @nestjs/graphql apollo-server-express graphql-tools graphql
+```
+
+Add resources related to a todo entity:
+
+```plaintext
+nest generate resource todo
+```
+- Choose `GraphQL (code first)`
+- Choose `Y` for CRUD
+
+---
+
+## Configure GraphQL in NestJS (code-first)
+
+Modify `/graphql-training/api/src/app.module.ts`
+
+```typescript
+import { Module } from '@nestjs/common';
+import { GraphQLModule } from '@nestjs/graphql';
+import { join } from 'path';
+import { TodoModule } from './todo/todo.module';
+import { AppController } from './app.controller';
+import { AppService } from './app.service';
+
+@Module({
+  imports: [
+    GraphQLModule.forRoot({
+      autoSchemaFile: join(process.cwd(), 'src/schema.graphql'),
+      playground: true,
+    }),
+    TodoModule,
+  ],
+  controllers: [AppController],
+  providers: [AppService],
+})
+export class AppModule {}
+```
+
+---
+
+## First GraphQL query
+
+> Context: `/graphql-training && docker-compose up -d`
+
+Open browser to access playground: http://localhost:3000/graphql
```

Comment:
This should be http://localhost:8000/graphql

Reply:
Thanks! I'll fix this after the session today. :-)

Reply:
Fixed!